validphys2/src/validphys/photon/structure_functions.py

```diff
@@ -0,0 +1,113 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pineappl
+from scipy.interpolate import RectBivariateSpline
+
+
+class StructureFunction(ABC):
+    """Abstract class for the DIS structure functions"""
+
+    @abstractmethod
+    def fxq(self, x, Q):
+        pass
+
+
+class InterpStructureFunction(StructureFunction):
+    """
+    Compute an interpolated structure function convoluting an FKtable
+    with a PDF.
+    """
+
+    def __init__(self, path_to_fktable, pdfs):
+        self.fktable = pineappl.fk_table.FkTable.read(path_to_fktable)
+        self.pdfs = pdfs
+        self.pdgid = int(pdfs.set().get_entry("Particle"))
```

Comment:
Does this also work if it's not a proton?

Reply:
When I made this comment I was thinking that it should always be a proton so there is no point in reading it from the `.info` files instead of hardcoding it while a typo in the `.info` file could cause problems here. On the other hand, I don't imagine there many such typos so I would also be fine if this were a test (though again, if you'd want to test this it may make more sense to implement it in the usual "vp-way" which may not really be worth it). 

I'll leave it up to you what you want to do :)

Reply:
In my answer I was not thinking to I typo in the info file, but instead to the fact that if someone tries to pass to the runcard a non proton PDF, e.g. MSHT20qed_nnlo_neutron (and therefore with pdgid != 2212) the code should stop it and give an error message.
But in any case I don't think that anyone will ever do something like that and the pdgid can even be harcoded

Reply:
I assumed that was your line of thinking, but I think we can assume the users (which are mostly us anyway) have some idea of what we're doing. I'm pretty sure this is not done for other pdfs that are used as input either, as (I assume) they're all parsed through the same function (as mentioned in my other comments about `parse_`) so I suppose this check could be added there but seems unnecessary to me.

Reply:
I agree with @RoyStegeman: we should make some assumptions and reduce the amount of checks.

Our main bound is to keep the code as simple as possible, to reduce the maintenance burden (we already have enough code and complex enough), so we can not sanitize any possible input from any possible user at each step. Make sane assumptions :)